### ISSUE TITLE
[bug] Error starting the client, incorrect database folder creation in Win32

### ISSUE THREAD

**Issue by kikoncuo** [ethereumjs/ethereumjs-client#25]
#### Description
When starting the client, if there is no chaindb folder in the project root, there is a crash

```Error: ENOENT: no such file or directory, mkdir 'D:\Personal\JS blockchain\Official\ethereumjs-client-master\chaindb\mainnet'
    at Object.fs.mkdirSync (fs.js:885:18)
    at targetDir.split.reduce (D:\Personal\JS blockchain\Official\ethereumjs-client-master\lib\chain\ChainManager.js:59:12)
    at Array.reduce (<anonymous>)
    at ChainManager._mkDirByPathSync (D:\Personal\JS blockchain\Official\ethereumjs-client-master\lib\chain\ChainManager.js:56:26)
    at ChainManager._initializeDatadir (D:\Personal\JS blockchain\Official\ethereumjs-client-master\lib\chain\ChainManager.js:71:12)
    at new ChainManager (D:\Personal\JS blockchain\Official\ethereumjs-client-master\lib\chain\ChainManager.js:15:10)
    at runClient (D:\Personal\JS blockchain\Official\ethereumjs-client-master\lib\index.js:24:12)
    at Object.<anonymous> (D:\Personal\JS blockchain\Official\ethereumjs-client-master\lib\index.js:27:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
```

This is because it tries to create the folder mainnet inside of the chaindb folder which doesn't exist

#### Possible Implementation

Add the empty folder to the project; will do this by creating the folder and inside a .placeholder file so the folder can be committed.

#### Reference Implementations / Links

Will post a PR as soon as done, it should be quick

**Comment by holgerd77** [comment]
Ah no, that's not a good way for implementation. This folder shouldn't be assumed to be there, since in the future, the folder should be configurable by CL. If a folder is not existing, it should be added recursively, I think I already added some base functionality for this in the ChainManager class.

**Comment by kikoncuo** [comment]
You are right, I fixed it as you proposed it, confirm if you haven't done it, I can create a PR

**Comment by holgerd77** [comment]
Yes, would be cool if you can open a PR

**Comment by kikoncuo** [comment]
Sorry for the delay, I've been swamped at work...

The problem existed only in windows: 
Path.sep tries to generate an array based on the elements in the path.
It takes in account OS, which means in POSIX it expects a / and in Win32 a \\, the provided path was / and in windows it didn't separate it correctly.

The problem is fixed by normalizing the path, good news is that the path library already implementsa solution for this, path.normalize(targetDir)

A PR has been created.

**Comment by vpulim** [comment]
Fixed in merged PR #37. Closing.
